File: binarysearch.com/2021/weekly_contests/week_44/problem_b.py
class Solution:
    def solve(self, s):
        l=[]
        c=0
        for i in s:
            if i == '(': 
                c+=1
                if c>len(l):
                    l.append(0)
            elif i==')': c-=1
            else: l[c-1]+=1
        return l

File: codeforces/2021/div3/round_702/problem_b.cpp
#include <bits/stdc++.h>
using namespace std;
 
void solve(){
	int n, a=0, b=0, c=0, t; cin>>n;
	for(int i=0; i<n; i++){
		cin>>t;
		if (t%3==0) a++;
		else if (t%3==1) b++;
		else c++;
	}
	n/=3;
	int r=0;
	
	for(int i=0; i<2; i++){
		if(a>n){ r+=(a-n); b+=(a-n); a=n;}
		if(b>n){r+=(b-n); c+=(b-n); b=n;}
		if (c>n){ r+=(c-n); a+=(c-n); c=n;}
	}
	cout<<r<<endl;
}
 
int main(){
	int t; cin>>t;
	while(t--) solve();
	return 0;
}
